client/src/features/visualizer-vivagraph/lib/helpers.ts

```diff
@@ -0,0 +1,22 @@
+import { IFeedBlockData } from "~models/api/nova/feed/IFeedBlockData";
+
+export const getBlockParents = (block: IFeedBlockData): string[] => {
+
+    // @ts-ignore
+    const parents = block.block?.body?.strongParents;
+
+    if (parents && parents.length) {
+        return parents;
+    }
+
+    // TODO confusing, because in interface method isBasic() exists, but in the implementation it does not
+    // if (block.block?.body?.isBasic()) {
+    //     return block.block?.body?.asBasic().strongParents;
+    // }
+
+    // if (block.block?.body?.isValidation()) {
+    //     return block.block?.body?.asValidation().strongParents;
+    // }
+
+    return [];
+}
```

Comment:
what about weakParents? maybe we can copy the Mario implementation in the three visualizer
```typescript
const parents: Parents = []
const blockStrongParents = (blockData.block.body as BasicBlockBody).strongParents ?? [];
const blockWeakParents = (blockData.block.body as BasicBlockBody).weakParents ?? [];
parents.push(...blockStrongParents, ...blockWeakParents)
return parents
```